esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3004,6 +3005,23 @@ void WsWuHelpers::submitWsWorkunit(IEspContext& context, IConstWorkUnit* cw, con
     const char *paramXml, IArrayOf<IConstNamedValue> *variables, IArrayOf<IConstNamedValue> *debugs, IArrayOf<IConstApplicationValue> *applications)
 {
     ensureWsWorkunitAccess(context, *cw, SecAccess_Write);
+
+#ifndef _NO_LDAP
+    CLdapSecManager* secmgr = dynamic_cast<CLdapSecManager*>(context.querySecManager());
```

Comment:
Fixed by wrapping view permissions test in if (secmgr && secmgr->getCheckViewPermissions()) block.